src/algebra/module/linear_map.lean

```diff
@@ -142,25 +177,25 @@ we can also add an instance for `add_comm_group.int_module`, allowing `z •` to
 -/
 class compatible_smul (R S : Type*) [semiring S] [has_scalar R M]
   [module S M] [has_scalar R M₂] [module S M₂] :=
-(map_smul : ∀ (f : M →ₗ[S] M₂) (c : R) (x : M), f (c • x) = c • f x)
+(map_smul : ∀ (fₗ : M →ₗ[S] M₂) (c : R) (x : M), fₗ (c • x) = c • fₗ x)
```

Comment:
Does this (and the lemmas depending on it) not generalize in a straightforward way to include `σ` as another argument to the typeclass? Could you either add a todo comment to the docstring about semilinear maps, or an explanation of why this typeclass doesn't make sense for them?

Reply:
Yes, not everything has been generalized to semilinear maps, this is one example. I added todos in the file-level docstring about this. I think this should be left to subsequent PRs, this one is already big enough!